File: CS/Form1.cs
using DevExpress.Utils.Menu;
using DevExpress.XtraPivotGrid;
using System;

namespace WinFormsPivot_App {
    public partial class Form1 : DevExpress.XtraEditors.XtraForm {
        public Form1() {
			InitializeComponent();
            pivotGridControl1.BestFit();
		}

		private void Form1_Load(object sender, EventArgs e) {
			// This line of code loads data into the 'nwindDataSet.ProductReports' table. You can move, or remove it, as needed.
			this.productReportsTableAdapter.Fill(this.nwindDataSet.ProductReports);
		}

        void pivotGridControl1_PopupMenuShowing(object sender, PopupMenuShowingEventArgs e) {
            if (e.MenuType == PivotGridMenuType.Header) {
                HandleHeaderShowMenu(e);
            }
        }

        private void HandleHeaderShowMenu(DevExpress.XtraPivotGrid.PopupMenuShowingEventArgs e) {
			if(!IsLastAreaField(pivotGridControl1, e.Field))
				CreateToggleTotalsItem(e);
		}

		private bool IsLastAreaField(PivotGridControl pivotGridControl, PivotGridField field) {
			for(int i = 0; i < pivotGridControl.Fields.Count; i++) {
				if(pivotGridControl.Fields[i].Visible && pivotGridControl.Fields[i].Area == field.Area 
					&& pivotGridControl.Fields[i].AreaIndex > field.AreaIndex)
					return false;
			}
			return true;
		}

        private void CreateToggleTotalsItem(PopupMenuShowingEventArgs e) {
			bool isTotalsVisible = e.Field.TotalsVisibility != PivotTotalsVisibility.None;
			DXMenuItem toggleTotalItem = new DXMenuItem(isTotalsVisible ? "Hide Totals" : "Show Totals", ToggleTotalsMenuItemClick);
			toggleTotalItem.BeginGroup = true;
			toggleTotalItem.Tag = e.Field;
			e.Menu.Items.Add(toggleTotalItem);
		}

		private void ToggleTotalsMenuItemClick(object sender, EventArgs e) {
			DXMenuItem item = (DXMenuItem)sender;
			PivotGridField field = (PivotGridField)item.Tag;
			if(field.TotalsVisibility == PivotTotalsVisibility.None)
				field.TotalsVisibility = PivotTotalsVisibility.AutomaticTotals;
			else
				field.TotalsVisibility = PivotTotalsVisibility.None;
		}
	}
}

File: VB/Form1.vb
Imports System
Imports System.ComponentModel
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.Utils.Menu
Imports DevExpress.XtraPivotGrid

Namespace Q134867

    Public Partial Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs)
            ' TODO: This line of code loads data into the 'nwindDataSet.ProductReports' table. You can move, or remove it, as needed.
            productReportsTableAdapter.Fill(nwindDataSet.ProductReports)
        End Sub

        Private Sub pivotGridControl1_PopupMenuShowing(ByVal sender As Object, ByVal e As DevExpress.XtraPivotGrid.PopupMenuShowingEventArgs)
            If e.MenuType Is DevExpress.XtraPivotGrid.PivotGridMenuType.Header Then
                Me.HandleHeaderShowMenu(e)
            End If
        End Sub

        Private Sub HandleHeaderShowMenu(ByVal e As DevExpress.XtraPivotGrid.PopupMenuShowingEventArgs)
            If Not IsLastAreaField(pivotGridControl1, e.Field) Then Me.CreateToggleTotalsItem(e)
        End Sub

        Private Function IsLastAreaField(ByVal pivotGridControl As PivotGridControl, ByVal field As PivotGridField) As Boolean
            For i As Integer = 0 To pivotGridControl.Fields.Count - 1
                If pivotGridControl.Fields(i).Visible AndAlso pivotGridControl.Fields(i).Area Is field.Area AndAlso pivotGridControl.Fields(i).AreaIndex > field.AreaIndex Then Return False
            Next

            Return True
        End Function

        Private Sub CreateToggleTotalsItem(ByVal e As DevExpress.XtraPivotGrid.PopupMenuShowingEventArgs)
            Dim isTotalsVisible As Boolean = e.Field.TotalsVisibility IsNot PivotTotalsVisibility.None
            Dim toggleTotalItem As DXMenuItem = New DXMenuItem(If(isTotalsVisible, "Hide Totals", "Show Totals"), AddressOf ToggleTotalsMenuItemClick)
            toggleTotalItem.BeginGroup = True
            toggleTotalItem.Tag = e.Field
            e.Menu.Items.Add(toggleTotalItem)
        End Sub

        Private Sub ToggleTotalsMenuItemClick(ByVal sender As Object, ByVal e As EventArgs)
            Dim item As DXMenuItem = CType(sender, DXMenuItem)
            Dim field As PivotGridField = CType(item.Tag, PivotGridField)
            If field.TotalsVisibility Is PivotTotalsVisibility.None Then
                field.TotalsVisibility = PivotTotalsVisibility.AutomaticTotals
            Else
                field.TotalsVisibility = PivotTotalsVisibility.None
            End If
        End Sub
    End Class
End Namespace
